app/src/main/java/com/sopt/dive/ui/main/search/SearchScreen.kt

```diff
@@ -0,0 +1,83 @@
+package com.sopt.dive.ui.main.search
+
+import androidx.compose.animation.core.EaseInOut
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.sopt.dive.R
+import com.sopt.dive.util.clickableWithoutRipple
+
+enum class ImgState { Collapsed, Expanded }
+
+@Composable
+fun SearchScreen(
+    modifier: Modifier = Modifier,
+) {
+    var isEnabled by remember { mutableStateOf(ImgState.Collapsed) }
+
+    val animateFloatAsState: Float by animateFloatAsState(
+        targetValue = if (isEnabled == ImgState.Collapsed) 0f else 180f,
+        animationSpec = tween(durationMillis = 2000, easing = EaseInOut)
+    )
+
+    Column(
+        modifier = modifier.fillMaxSize(),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center
+    ) {
+        Box(
+            modifier = Modifier
+                .size(width = 300.dp, height = 500.dp)
+                .graphicsLayer(
+                    rotationY = animateFloatAsState,
+                    cameraDistance = 12f,
+                )
+                .clickableWithoutRipple(
+                    onClick = {
+                        isEnabled = when (isEnabled) {
+                            ImgState.Collapsed -> ImgState.Expanded
+                            ImgState.Expanded -> ImgState.Collapsed
+                        }
+                    }
+                )
+        ) {
+            if (animateFloatAsState > 90) {
+                Image(
+                    painter = painterResource(id = R.drawable.img_king),
+                    contentDescription = "king",
+                    modifier = Modifier
+                        .fillMaxSize(),
+                )
+            } else {
+                Image(
+                    painter = painterResource(id = R.drawable.img_card_back3),
+                    contentDescription = "card",
+                    modifier = Modifier
+                        .fillMaxSize(),
+                )
```

Comment:
둘의 차이는 id, contentDescription 둘로 보여요.

```suggestion
                Image(
                    painter = painterResource(id = if (animateFloatAsState > 90) R.drawable.img_king else  R.drawable.img_card_back3),
                    contentDescription = if (animateFloatAsState > 90) "king" else "card,
                    modifier = Modifier
                        .fillMaxSize(),
                )
```
로 변경하면 코드가 더 짧아지는데 어떻게 생각하시나용?

또한 Image에 modifier를 사용해준다면  Box가 없어도 될 것 같아요.

Reply:
확실히 필요없는 Box를 없애고, 조건문을 추가하는게 좋아보이네요!!
아예 카드에 대한 dataClass를 만들어서 관리해봐야겠어요.